File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    #전체 논문 중 h회 이상 인용된 논문이 h편 이상
    #역순으로 정렬
    #index를 증가시키면서
    #index + 1보다 
    citations.sort(reverse = True)
    
    for index in range(len(citations)):
        subscribe_count = citations[index]
        if index + 1 > subscribe_count:
            return  index
        
    return len(citations)

File: 프로그래머스/2/42747. H－Index/H－Index.js
// h번 이상 인용된 논문이 h편 이상이면 h -> 최댓값

// 내림차순 정렬 후
// 처음부터 나까지의 개수가 나의 값보다 크거나 같은 값의 최대값

function solution(citations) {
    citations.sort((a, b) => b - a);
    
    let count = 0;
    
    for(let i = 0; i < citations.length; i++){
        const 인용횟수 = citations[i];
        const 논문개수 = i + 1;
        
        if(논문개수 > 인용횟수){
            return count;
        }else{
            count = 논문개수;
        }
    }
    
    return count;
}